packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts

```diff
@@ -22,8 +22,24 @@ export class ClusterResourceHandler extends ResourceHandler {
   constructor(eks: EksClient, event: ResourceEvent) {
     super(eks, event);
 
-    this.newProps = parseProps(this.event.ResourceProperties);
-    this.oldProps = event.RequestType === 'Update' ? parseProps(event.OldResourceProperties) : { };
+    function patchEndpointAccess(props: any) {
+
+      // this is weird but these boolean properties are passed by CFN as a string, and we need them to be booleanic for the SDK.
+      // Otherwise it fails with 'Unexpected Parameter: params.resourcesVpcConfig.endpointPrivateAccess is expected to be a boolean'
+
+      if (typeof(props.resourcesVpcConfig?.endpointPrivateAccess) === 'string') {
+        Object.assign(props.resourcesVpcConfig, { endpointPrivateAccess: props.resourcesVpcConfig.endpointPrivateAccess === 'true' });
+      }
+
+      if (typeof(props.resourcesVpcConfig?.endpointPublicAccess) === 'string') {
+        Object.assign(props.resourcesVpcConfig, { endpointPublicAccess: props.resourcesVpcConfig.endpointPublicAccess === 'true' });
+      }
+
+      return props;
+    }
+
+    this.newProps = patchEndpointAccess(parseProps(this.event.ResourceProperties));
```

Comment:
Can you merge `patchEndpointAccess` to `parseProps`?

Reply:
👍